runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/ProcessInstanceEvent/ProcessInstanceEvent.reducers.ts

```diff
@@ -18,3 +20,23 @@ export const reduceCrumbs = (
         index + 1
     );
 };
+
+export const updateSingleActiveLoopEvent = (state: ProcessInstanceEventState, action: PayloadAction<IProcessInstanceLoopEvent>) => {
+    if (state.all.nestedEvents[action.payload.loopId]) {
+        if (!state.all.nestedEvents[action.payload.loopId][action.payload.iterationNumber]) {
+            state.all.nestedEvents[action.payload.loopId][action.payload.iterationNumber] = [action.payload];
+            return;
+        }
+
+        if (state.all.nestedEvents[action.payload.loopId][action.payload.iterationNumber].some((loopEvent) => loopEvent.id === action.payload.id)) {
+            state.all.nestedEvents[action.payload.loopId][action.payload.iterationNumber] = state.all.nestedEvents[action.payload.loopId][action.payload.iterationNumber]
+                .map((loopEvent) => {
+                    if (loopEvent.id === action.payload.id) return action.payload;
+                    return loopEvent;
+                });
+            return;
+        }
+
+        state.all.nestedEvents[action.payload.loopId][action.payload.iterationNumber].push(action.payload);
+    }
+};
```

Comment:
assign `state.all.nestedEvents[action.payload.loopId][action.payload.iterationNumber]` to a const - it will simplify a lot your code.

Reply:
can only assign the value access part of this - otherwise redux toolkit doesn't understand the direct mutations in array